### DISCUSSION TITLE
How to remap enter to a different command to execute

### DISCUSSION THREAD

**Discussion by bernardassan** [junegunn/fzf#2897] [Q&A] [answered]
I use fzf as my launcher in sway an have the command below in my launcher.zsh
```
#unbind isn't evaluated immediately until an event or key stroke is executed and rebind is used to bind after unbinding
FZF_DEFAULT_COMMAND='find /usr/share/applications -name "*.desktop" | xargs basename -s ".desktop"' \
        fzf --no-multi --height 100% --layout reverse-list --margin 3% --padding 3% --info hidden \
        --bind 'ctrl-d:reload(sleep 0.01;find /usr/share/applications -name "*.desktop" | xargs basename -s ".desktop")+change-prompt(desktop > )+rebind(enter)+unbind(alt-enter)' \
        --bind 'enter:execute(swaymsg -t command exec gtk-launch {})+abort' \
        --bind 'ctrl-e:reload(printenv PATH | xargs -d : -I {} -r find -L {} -maxdepth 1 -type f -executable -printf "%f\n" 2>/dev/null| sort -u)+change-prompt(executable > )+rebind(alt-enter)+unbind(enter)' \
        --bind 'alt-enter:execute(swaymsg -t command exec kitty {})+abort' \
        --prompt 'desktop > ' \
        --header-first --header "					sway app launcher"
```
How can I rebind `enter` to execute `swaymsg -t command exec kitty {}` when in the `ctrl-e` mode (executable) and `swaymsg -t command exec gtk-launch {}` when in `ctrl+d` mode (desktop)

if that isn't possible, Is the a way I can prevent `alt-enter` from working in the default fzf command mode (the initial interface command loaded with FZF_DEFAULT_COMMAND) because with what i have now , I am able to disable the wrong key from been pressed in a different mode but only after either `ctrl-e` or `ctrl-d` has been executed. In default mode, both `enter` and `alt-enter` can be pressed and that leads to some surprisingly unexpected results. I have read the man pages and saw the `--expect` option , I thought that is what I needed but It doesn't do what I want.

**Comment by tg0h** [comment, marked as answer]
Have a look at https://github.com/junegunn/fzf/wiki/Examples#dnf

The example does not unbind and rebind `enter`. Rather, it uses a single `enter` command, stores the current state (whether in `ctrl-e` or `ctrl-d` mode) in a temp file and then uses that to determine what `enter` should do.

**Comment by bernardassan** [comment]
@tg0h Thank you for the pointer, The examples did help.
